test/scenario/dg-upgrade-scenarios.t.sol

```diff
@@ -0,0 +1,921 @@
+            extendEmergencyProtectionCallsBuilder.addCall(
+                address(_timelock), abi.encodeCall(_timelock.setGovernance, (_timelock.getEmergencyGovernance()))
+            );
+
+            // Set emergency protection end date
+            extendEmergencyProtectionCallsBuilder.addCall(
+                address(_timelock), abi.encodeCall(_timelock.setEmergencyProtectionEndDate, (emergencyModeEndsAfter))
+            );
+
+            // Set new emergency mode duration for a year
+            extendEmergencyProtectionCallsBuilder.addCall(
+                address(_timelock), abi.encodeCall(_timelock.setEmergencyModeDuration, (emergencyModeDuration))
+            );
+
+            // Reset emergency protection committees
+            extendEmergencyProtectionCallsBuilder.addCall(
+                address(_timelock),
+                abi.encodeCall(_timelock.setEmergencyProtectionActivationCommittee, (emergencyActivationCommittee))
+            );
+
+            // Reset emergency protection execution committee
+            extendEmergencyProtectionCallsBuilder.addCall(
+                address(_timelock),
+                abi.encodeCall(_timelock.setEmergencyProtectionExecutionCommittee, (emergencyExecutionCommittee))
+            );
+
+            uint256 proposalId = _submitProposalByAdminProposer(
+                extendEmergencyProtectionCallsBuilder.getResult(), "Extend emergency protection"
+            );
+            _assertProposalSubmitted(proposalId);
+            _assertSubmittedProposalData(proposalId, extendEmergencyProtectionCallsBuilder.getResult());
+
+            _wait(_getAfterSubmitDelay());
+
+            _assertCanSchedule(proposalId, true);
+            _scheduleProposal(proposalId);
+
+            _wait(_getAfterScheduleDelay());
+
+            //Proposal can't be executed, because emergency mode is active
+            vm.expectRevert(abi.encodeWithSelector(EmergencyProtection.UnexpectedEmergencyModeState.selector, true));
+            _executeProposal(proposalId);
+
+            //Emergency committee executes the proposal
+            _emergencyExecute(proposalId);
+            _assertProposalExecuted(proposalId);
+
+            // Check that emergency protection is extended
+            assertEq(_timelock.getEmergencyActivationCommittee(), emergencyActivationCommittee);
+            assertEq(_timelock.getEmergencyExecutionCommittee(), emergencyExecutionCommittee);
+            assertTrue(_timelock.isEmergencyProtectionEnabled());
+            assertFalse(_timelock.isEmergencyModeActive());
+
+            // Check malicious proposal was cancelled
+            _assertProposalCancelled(maliciousProposalId);
+        }
+
+        _step("7. DAO operates as usually");
+        {
+            ExternalCall[] memory regularStaffCalls = _getMockTargetRegularStaffCalls();
+            uint256 proposalId = _submitProposalByAdminProposer(regularStaffCalls, "DAO performs regular stuff");
```

Comment:
It works only because both the DG proposer and the `TimelockedGovernance` proposer are set to the `Voting` contract. It’s better to use the `_submitProposal()` helper here and explicitly pass `Voting` as the proposer.